### PR TITLE
Task/setpatch

### PR DESCRIPTION
**Pull request by vanosg**
Found by: maimizuno
Patch by: Geo
Fixes: #337 

One-line summary: Revise eggdrop internal versioning system

Additional description (if needed):
Use version.h to set version/patchlevel via misc/setpatch. Each time setpatch is run, the patchlevel digits of egg_numver is incremented by 1, not to exceed 99. misc/newversion was also updated to incorporate this. Under this system, misc/newversion should be run just prior to releasing a Release Candidate. Using 1.8.1 as an example, newversion sets numver to 1080100.  Using setpatch to update for RC (as part of releaseprep), numver will be updated to 1080101. Each new RC will increment by 1, and the stable will become RCX+1 (So, assuming 2 RCs, 1080103). Patches in the develop branch beyond stable will increment the patchlevel digits (last two), until newversion is called again to prepare the next sub-release and the cycle repeats. 

Test cases demonstrating functionality (if applicable):

### CODE REVIEW

File: misc/newversion [hunk 1]
```diff
@@ -43,16 +42,16 @@ umask 022
 
 OLD_EGGVERSION=$(grep AC_INIT configure.ac | sed -e 's/AC_INIT(\[Eggdrop\],\[//g' -e 's/\],\[bugs@eggheads.org\])//g')
 NEW_EGGVERSION=$(echo $OLD_EGGVERSION | cut -d. -f1-2).$(($(echo $OLD_EGGVERSION | cut -d. -f3) + 1))
-NEW_NUMVERSION=$(($(grep 'egg_numver =' src/main.c | sed -e 's/.*= //' -e 's/;$//') + 100))
+NEW_NUMVERSION=$(($(grep -E '^#define EGG_NUMVER.*' src/version.h |cut -d " " -f 3|sed -e 's/..$//' -e 's/$/00/') + 100))
```
Comment (Cizzle) [inline misc/newversion]: Why not use "sed -e 's/..$/00/'"? And I don't see any ERE's in that regex needing "grep -E"?

File: misc/setpatch [hunk 2]
```diff
@@ -0,0 +1,110 @@
+#! /bin/bash
+#
+# addpatch - modifies src/patch.h
```
Comment (Cizzle) [inline misc/setpatch]: Needs update.

File: misc/setpatch [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+#
+# As a special exception to the GNU General Public License, if you
+# distribute this file as part of a program that contains a
+# configuration script generated by Autoconf, you may include it under
+# the same distribution terms that you use for the rest of that program.
+
+show_usage() {
+	echo "Usage: `basename $0` [-hs] <patchname>"
+	echo ""
+	echo "  patchname  - Name of patch to add."
```
Comment (Cizzle) [inline misc/setpatch:30]: Needs update too I guess?

File: misc/setpatch [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+else
+	patch_name=$1
+fi
+
+if test ! -f src/main.c; then
+	echo "You are not in the Eggdrop root directory."
+	exit 1
+fi
+
+oldpatch_name=$(cat src/version.h |grep -E '^#define EGG_PATCH.*'|cut -d "\"" -f 2)
```
Comment (Cizzle) [inline misc/setpatch]: Save a process, kill the cat.

File: misc/setpatch (hunk 2 above)
Comment (vanosg) [reply misc/setpatch]: Updated

File: misc/setpatch (hunk 3 above)
Comment (vanosg) [reply misc/setpatch:30]: This is accurate for usage

File: misc/setpatch (hunk 4 above)
Comment (vanosg) [reply misc/setpatch]: Updated

File: misc/setpatch [hunk 5]
```diff
@@ -0,0 +1,111 @@
+#! /bin/bash
+#
+# addpatch - generates src/version.h
```
Comment (Cizzle) [inline misc/setpatch:3]: "setpatch"

**Comment by Cizzle** [comment]
src/Makefile.in still has patch.h ref.

**Pull request merged**